core/src/main/java/org/axonframework/eventsourcing/eventstore/jdbc/JdbcEventStorageEngine.java

```diff
@@ -261,6 +261,38 @@ protected void storeSnapshot(DomainEventMessage<?> snapshot, Serializer serializ
                                    ))));
     }
 
+    @Override
+    public TrackingToken createHeadToken() {
+        String sql = "SELECT max(" + schema.globalIndexColumn() + ") FROM " + schema.domainEventTable();
+        Long index = transactionManager.fetchInTransaction(
+                () -> executeQuery(getConnection(),
+                                   connection -> connection.prepareStatement(sql),
+                                   resultSet -> resultSet.next() ? resultSet.getObject(1, Long.class) : null,
+                                   e -> new EventStoreException("Failed to get head token")));
+        return Optional.ofNullable(index)
+                       .map(seq -> GapAwareTrackingToken.newInstance(seq, Collections.emptySet()))
+                       .orElse(null);
+    }
+
+    @Override
+    public TrackingToken createTokenAt(Instant dateTime) {
+        String sql = "SELECT max(" + schema.globalIndexColumn() + ") FROM " + schema.domainEventTable() + " WHERE "
```

Comment:
This query doesn't seem to match the requirement. This method should return a token that will give you all the events that have a timestamp on or after the given dateTime. This _may_ mean there are events in between that have a lower timestamp, because creation and publication order are not identical.
The query should be `SELECT min(...) - 1 FROM ... WHERE timestamp >= dateTime`